File: Dart/3096.dart
import 'dart:io';
import 'dart:math';

double log10(num x) => log(x) / ln10;

void main() {
  int n = int.parse(stdin.readLineSync());
  
  double r = n*log10(n/e) + log10(2*pi*n)/2.0;
  print("${r.floor() + 1}");
}

File: Java/3096.java
import java.io.IOException;
import java.util.Scanner;

public class Main {
public static void main(String[] args) throws IOException {
        Scanner scanner = new Scanner(System.in);
        
        double n = scanner.nextInt();
        System.out.println((int) Math.floor(n*Math.log10(n/Math.E) + Math.log10(2*Math.PI*n)/2.0) + 1);
    }
}